src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java

```diff
@@ -0,0 +1,238 @@
+        }
+    }
+
+    private List<RevCommit> getAllCommits() throws Exception {
+        Iterable<RevCommit> commits = git.log().add(repository.resolve("HEAD")).call();
+        int count = 0;
+        List<RevCommit> commitList = new ArrayList<>();
+        for (RevCommit commit : commits) {
+            System.out.println("LogCommit: " + commit +
+                    ", name: " + commit.getName() + ", id: " + commit.getId().getName());
+            count++;
+            commitList.add(commit);
+        }
+        System.out.println(count);
+        return commitList;
+    }
+
+    private void generateTwoCommitDiffPatch(RevCommit commitOld, RevCommit commitNew) throws Exception {
+        final ObjectReader reader = git.getRepository().newObjectReader();
+        final CanonicalTreeParser oldTreeIter = new CanonicalTreeParser();
+        oldTreeIter.reset(reader, commitOld.getTree());
+        final CanonicalTreeParser newTreeIter = new CanonicalTreeParser();
+        newTreeIter.reset(reader, commitNew.getTree());
+
+        String dirName = new File(".").getAbsoluteFile().getParent() + "/DiffReport";
+        String subDirName = getSimpleRepoName() + "-" + commitNew.getName().substring(0, 7);
+        String destDirName = dirName + "/" + subDirName;
+        File destDirFile = new File(destDirName);
+        if (destDirFile.exists()) {
+            destDirFile.delete();
+        }
+        boolean succ = new File(destDirName).mkdir();
+        if (succ) {
+            File patchFile = new File(dirName, "patch.txt");
+            final PrintStream ps = new PrintStream(new FileOutputStream(patchFile));
+            final DiffFormatter diffFormatter = new DiffFormatter(ps);
+            diffFormatter.setRepository(git.getRepository());
+            final List<DiffEntry> entries = diffFormatter.scan(oldTreeIter, newTreeIter);
+            diffFormatter.format(entries);
+            diffFormatter.close();
+
+            checkout(commitNew.getName());
+            File reportDir = generate();
+            copyFolder(reportDir, new File(destDirName));
+
+            final DateFormat format = new SimpleDateFormat("yyyyMMddHHmmss", Locale.getDefault());
+            final String patchFileName = String.format("%s-patch-%s-%s-%s.txt", getSimpleRepoName(),
+                    commitOld.getName().substring(0, 7), commitNew.getName().substring(0, 7), format.format(new Date()));
+            if (patchFile.renameTo(new File(destDirName, patchFileName))) {
+                System.out.println("File is moved successful!");
+            }
+        }
+    }
+
+    private void checkout(String commitName) throws Exception {
+        git.checkout().setName(commitName).call();
+    }
+
+    private File generate()
+            throws InterruptedException, IOException {
+        final Process process = new ProcessBuilder()
+                .directory(new File(testerPath))
+                .command(
+                        "groovy", "diff.groovy",
```

Comment:
you do not need diff.groovy

please use inner generator of diff report based on 2 checkstyle xml reports https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/diff.groovy#L209

Reply:
done, understand